### PR TITLE
document testing

### LINKED ISSUE #69: document testing regimen
/assign
/sig k8s-infra

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes #69

### CODE REVIEW

File: cmd/archeio/docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Changes to archeio are auto-deployed to registry-staging.k8s.io and NOT to
+registry.k8s.io. registry.k8s.io serves stable releases.
+
+The instance at registry-staging.k8s.io has [kops] cluster CI running in AWS 
+pointed at it to validate pulling from AWS. The kops clusters run in random
+regions.
+
+This E2E CI should be consulted before promoting code to stable release + registry.k8s.io.
+
+Results are visible in [testgrid] at: https://testgrid.k8s.io/sig-k8s-infra-oci-proxy#Summary
```
Comment (BenTheElder) [inline cmd/archeio/docs/testing.md:47]: fun fact: currently failing https://testgrid.k8s.io/sig-k8s-infra-oci-proxy#kops-grid-gcr-mirror-canary

we should get alerts to the SIG, if possible. not sure if we have an alias for that yet.

File: cmd/archeio/docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+The integration tests run the actual application `main()`, and pull image(s)
+through a running instance using [crane].
+
+`make integration` runs only integration tests.
+
+Because our CI runs primarily in GCP, this currently covers pulling from outside AWS.
+
+## E2E Testing
+
+Changes to archeio are auto-deployed to registry-staging.k8s.io and NOT to
```
Comment (ameukam) [inline cmd/archeio/docs/testing.md:38]:
```suggestion
Changes to archeio are auto-deployed to registry-sandbox.k8s.io and NOT to
```

unless we plan to have `registry-staging.k8s.io` ?

File: cmd/archeio/docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+registry.k8s.io. registry.k8s.io serves stable releases.
+
+The instance at registry-staging.k8s.io has [kops] cluster CI running in AWS
```
Comment (ameukam) [inline cmd/archeio/docs/testing.md:41]:
```suggestion
The instance at registry-sandbox.k8s.io has [kops] cluster CI running in AWS 
```

same

File: cmd/archeio/docs/testing.md (hunk 1 above)
Comment (ameukam) [reply cmd/archeio/docs/testing.md:47]: You can use `k8s-infra-oci-proxy-admins@kubernetes.io' for alerting.

File: cmd/archeio/docs/testing.md (hunk 2 above)
Comment (BenTheElder) [reply cmd/archeio/docs/testing.md:38]: ACK 🙃

File: cmd/archeio/docs/testing.md (hunk 1 above)
Comment (BenTheElder) [reply cmd/archeio/docs/testing.md:47]: let's do that.

Comment (BenTheElder) [reply cmd/archeio/docs/testing.md:47]: actually, scratch that, we only care when we're going to promote to production, and we're all oversubscribed.

File: cmd/archeio/docs/testing.md (hunk 3 above)
Comment (BenTheElder) [reply cmd/archeio/docs/testing.md]: done

File: cmd/archeio/docs/testing.md (hunk 2 above)
Comment (BenTheElder) [reply cmd/archeio/docs/testing.md]: done.

**Pull request merged**